src/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -398,6 +400,34 @@ public void Timeout_TooShort_AllPendingOperationsCanceled()
             }
         }
 
+        [Fact]
+        public async Task SendAsync_UserAgent_CorrectlyWritten()
+        {
+            string userAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_3) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/63.0.3239.18 Safari/537.36";
+            string requestUri = "http://[::1234]";
+            bool connectionAccepted = false;
+
+            await LoopbackServer.CreateClientAndServerAsync(async proxyUri =>
+            {
+                using (HttpClientHandler handler = CreateHttpClientHandler())
+                using (var client = new HttpClient(handler))
+                {
+                    handler.Proxy = new WebProxy(proxyUri);
+                    var message = new HttpRequestMessage(HttpMethod.Post, requestUri);
+                    message.Headers.TryAddWithoutValidation("User-Agent", userAgent);
+                    try { await client.SendAsync(message).ConfigureAwait(false); }
```

Comment:
Why eat the exception?

Reply:
If the test is changed to just use the loopback server as the destination, then the request will succeed and no exception needs to be eaten.  The side-affect of using the loopback server is being able to inspect the string[] of request headers which is what the test is validating for the 'User-Agent' request header test.